src/js/middleware.js

```diff
@@ -0,0 +1,23 @@
+"use strict";
+// Base grade for express middleware modules...
+//
+// The express module will automatically attempt to wire in anything with this gradeName into itself
+
+var fluid = fluid || require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.express.middleware");
+
+gpii.express.middleware.complainAboutMissingFunction = function () {
+    fluid.fail(new Error("You must implement your own getMiddleware invoker."));
+};
+
+fluid.defaults("gpii.express.middleware", {
+    gradeNames: ["fluid.eventedComponent", "fluid.modelRelayComponent", "autoInit"],
```

Comment:
I had to add the grade to one of the derived components (the "counter" middleware), but the parent component itself is fine without that grade.